cloudevents/formats/json-format.md

```diff
@@ -143,7 +143,8 @@ If the implementation determines that the type of data is `Binary`, the value
 MUST be represented as a [JSON string][json-string] expression containing the
 [Base64][base64] encoded binary value, and use the member name `data_base64` to
 store it inside the JSON representation. If present, the `datacontenttype` MUST
-reflect the format of the original binary data.
+reflect the format of the original binary data. If the `datacontenttype` is
+unspecified, processing MUST NOT assume a default of `application/json`.
```

Comment:
I guess you mention it there already - I'm not sure it needs to go here too since this section is about serializing not "processing". right?

Reply:
I used "processing" as that's what's used in the paragraph below. "If the datacontenttype is unspecified, processing SHOULD proceed as if the datacontenttype had been specified explicitly as application/json." Happy to use another word if we can find a suitable one though.

Reply:
interesting. Seems like perhaps both should just be in the deserialize section

Reply:
We definitely need it in the serialization section as well - because serializing it *does* (or at least can) add the implicit datacontenttype in the HTTP response (or whatever).

Reply:
https://github.com/cloudevents/sdk-csharp/issues/200 is the C# issue of "whoops, we messed up here".

Here's the C# code which is broken, by unconditionally adding in `"datacontenttype": "application/json"` into the document if the event has data but no `datacontenttype`.

https://github.com/cloudevents/sdk-csharp/blob/409982e9e97a8a450b1bfbbbc2a16a8c08c78f62/src/CloudNative.CloudEvents.NewtonsoftJson/JsonEventFormatter.cs#L449-L453

Reply:
(I'd expected that to actually show the code, but it doesn't for some reason. Ah well - follow the link to see my failure...)